src/HW_models/NRF_RTC.c

```diff
@@ -125,15 +124,52 @@ static void time_of_1_counter_wrap(int rtc, bs_time_t *us, uint16_t *frac) {
   *us = counter_to_time((uint64_t)RTC_COUNTER_MASK + 1, rtc, frac);
 }
 
+/**
+ * Convert a counter value to absolute time.
+ * @note To obtain a time delta, use @ref counter_to_time instead
+ */
+static bs_time_t counter_to_abs_time(int rtc, uint32_t rtc_counter) {
+  uint16_t frac;
+  bs_time_t bs_time = RTC_counter_startT[rtc]
+                      + counter_to_time(rtc_counter, rtc, &frac);
+
+  bs_time_t now = tm_get_hw_time();
+
+  if (bs_time <= now) {
+    bs_time_t t_us;
+    uint16_t t_frac;
+
+    time_of_1_counter_wrap(rtc, &t_us, &t_frac);
+
+    do {
```

Comment:
Note: Using the TICK events will trigger a lot of events in this code. Which will all the time calling this function. And if the RTC is not started/stopped regularly will unfortunately lead to slowing simulations as RTC_counter_startT is more and more in the past.
You don't necessarily need to fix it now, but if you don't, please add a big warning/TODO at the top, telling that the code should be changed with care to not make the RTC drift due to the LF clock period not being an integer number of microseconds)

Reply:
Yes, you are right. Using this event slows down simulation. I've added a trace to `nrf_rtc_regw_sideeffect_EVTENSET()` to make this clear to the user of this model.